src/main/java/org/distributed/systems/chord/service/StorageService.java

```diff
@@ -0,0 +1,28 @@
+package org.distributed.systems.chord.service;
+
+import java.io.Serializable;
+import java.util.HashMap;
+import java.util.Map;
+
+public class StorageService {
+
+    private static final StorageService instance = new StorageService();
+    private Map<String, Serializable> valueStore;
+
+    public static StorageService getInstance() {
```

Comment:
Oh wait I see you're using it in the MemCachedHandler class. I'm not sure I guess we can keep it for now. I think in the akka paradigm it might be normally handled by the MemCachedHandler actor that would send a message to the node (internally).

Reply:
Okay, how about we use the MemCachedHandler as an `KeyValueStoreActor`, which provides both the TCP Interface and the Akka Communication (Node forwards all KeyValue messages to that actor). Then the node is soley responsible for handling the Chord Network and incoming requests, that are passed to the new KeyValueStoreActor. The KeyValueStore actor then holds the StorageService